scipy/stats/_hypotests.py

```diff
@@ -663,3 +666,353 @@ def somersd(x, y=None):
         raise ValueError("x must be either a 1D or 2D array")
     d, p = _somers_d(table)
     return SomersDResult(d, p, table)
+
+
+def _compute_log_combinations(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = gammaln(np.arange(n + 1) + 1)
+    return gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+BarnardExactResult = make_dataclass(
+    "BarnardExactResult", [("statistic", float), ("pvalue", float)]
+)
+
+
+def barnard_exact(table, alternative="two-sided", pooled=True, n=32):
+    r"""Perform a Barnard exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+        * 'two-sided'
+        * 'less': one-sided
+        * 'greater': one-sided
+
+    pooled : bool, optional
+        Whether to compute score statistic with pooled variance (Student's
+        t-test) or unpooled variance (Welch's t-test). Default is ``True`` :
+        Statistic test is computed using pooled variance.
+
+    n : int, optional
+        Number of sampling points used in the construction of the sampling
+        method. Note that this argument must be a power of two since we are
+        using Sobol's sampling method `scipy.stats.qmc.Sobol`.
```

Comment:
This would be ok. What I meant is that the correct orthographe has a `'`. So it is **Sobol'** and not *Sobol*.
```suggestion
        using `scipy.stats.qmc.Sobol`.
```